dialog.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017, djcj <djcj@gmx.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <FL/Fl.H>
+#include <FL/Fl_Button.H>
+#include <FL/Fl_Check_Button.H>
+#include <FL/Fl_Double_Window.H>
+#include <FL/Fl_PNG_Image.H>
+
+#include <algorithm>
+#include <sstream>
+#include <vector>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "desktopenvironments.h"
+#include "dialog_images.h"
+
+const char *msg_launch      = "\nLaunch app\n\n";
+const char *msg_menu        = "\nCreate menu entry\nand launch app";
+const char *msg_checkbox    = " Don't show this message again";
+
+const char *msg_launch_de   = "\nStarte App\n\n";
+const char *msg_menu_de     = "\nMenüeintrag erstellen\nund App starten";
+const char *msg_checkbox_de = " Dieses Fenster nicht erneut anzeigen";
+
+enum {
+  RET_LAUNCH_APP = 0,
+  RET_MENU_ENTRY = 1,
+  RET_DISABLE_MESSAGE = 2,
+  RET_CLOSE_WINDOW = 3
+};
+
+Fl_Double_Window *win;
+int rv = 0;
+bool checkbutton_set = false;
+
+std::vector<std::string> splitString(const std::string &str, const char delimiter = ' ')
+{
+  std::vector<std::string> result;
+  std::stringstream ss(str);
+  std::string item;
+
+  if (str.empty())
+  {
+    ss << " ";
+  }
+
+  while (std::getline(ss, item, delimiter))
+  {
+    result.push_back(item);
+  }
+  return result;
+}
+
+void get_system_font(std::string &font)
+{
+  std::vector<std::string> splitted;
+  std::stringstream ss;
+  int vsize;
+
+  auto *desktopEnvironment = IDesktopEnvironment::getInstance();
```

Comment:
Didn't you say you wanted to keep it C++98? Type inference has been introduced in C++11. You might want to rewrite that line in case this is still a requirement for you.